state/allwatcher_internal_test.go

```diff
@@ -2053,7 +2139,9 @@ func (s *allModelWatcherStateSuite) TestStateWatcher(c *gc.C) {
 			HasVote:   false,
 			WantsVote: false,
 		},
-	}})
+	}}
+
+	checkDeltasEqual(c, deltas, expectedDeltas)
```

Comment:
Looks like you came to a more accurate test for correctness.

Something that occurred to me in the weekend was that if the map conversion in `checkDeltas` is compressing entries so as to obfuscate the true sequence of events, then one solution is to check it more frequently in-between invoking the calls that cause delta emission.